modules/ding_webtrekk/ding_webtrekk.module

```diff
@@ -0,0 +1,101 @@
+  $script = 'window._ti = window._ti || {};';
+
+  $title = drupal_get_title();
+  if ($title) {
+    $script .= "window._ti['pageTitle'] = '$title';";
+  }
+  $domain = variable_get('webtrekk_ti_domain', '');
+  $id = variable_get('webtrekk_ti_id', '');
+
+  $page['page_bottom']['webtrekk']['#attached']['js'] = [
+    [
+      'data' => $script,
+      'type' => 'inline',
+    ],
+    [
+      'data' => 'window._tiConfig = window._tiConfig || {
+        tiDomain: ' . "'$domain'," .
+        'tiId: ' . "'$id'," .
+        'option: {}
+      };
+/** start TagIntegration loader  */
+(function(c,d,a,f){c.wts=c.wts||[];var g=function(b){var a="";b.customDomain&&b.customPath?a=b.customDomain+"/"+b.customPath:b.tiDomain&&b.tiId&&(a=b.tiDomain+"/resp/api/get/"+b.tiId+"?url="+encodeURIComponent(c.location.href)+"&v=5");if(b.option)for(var d in b.option)a+="&"+d+"="+encodeURIComponent(b.option[d]);return a};if(-1===d.cookie.indexOf("wt_r=1")){var e=d.getElementsByTagName(a)[0];a=d.createElement(a);a.async=!0;a.onload=function(){if("undefined"!==typeof c.wt_r&&!isNaN(c.wt_r)){var b=new Date,a=b.getTime()+1E3*parseInt(c.wt_r);b.setTime(a);d.cookie="wt_r=1;path=/;expires="+b.toUTCString()}};a.onerror=function(){"undefined"!==typeof c.wt_mcp_hide&&"function"===typeof c.wt_mcp_hide.show&&(c.wt_mcp_hide.show(),c.wt_mcp_hide.show=function(){})};a.src="//"+g(f);e.parentNode.insertBefore(a,e)}})(window,document,"script",_tiConfig);
+/** end TagIntegration loader */',
+      'type' => 'inline',
+      'scope' => 'footer',
+    ],
+  ];
+}
+
+/**
+ * Implements hook_panels_pane_content_alter().
+ */
+function ding_webtrekk_panels_pane_content_alter($content, $pane, $args, $contexts) {
+  if ($pane->type == 'search_result' && $pane->configuration['type'] == 'ting_search') {
+    if ($search_result = drupal_static('ting_search_results', FALSE)) {
+      $script = 'window._ti = window._ti || {};';
+      // Pass page title to the script.
+      if (!empty($content->original_title)) {
+        $script .= "\nwindow._ti['pageTitle'] = '$content->original_title';";
+      }
+
+      // Pass search options to the script.
+      $search_key = $search_result->search_key;
+      $count = $search_result->numTotalObjects;
+      $script .= "\nwindow._ti['OSS'] = '$search_key';\nwindow._ti['OSSr'] = '$count';";
+
+      drupal_add_js($script, array(
+        'type' => 'inline',
+      ));
+    }
+  }
+}
+
+/**
+ * Implements hook_views_post_execute().
+ */
+function ding_webtrekk_views_post_execute(&$view) {
+  $count = $view->total_rows;
+  if ($view->name == 'ding_multiple_search' && $count > 0) {
+    $script = 'window._ti = window._ti || {};';
+
+    $search_key = array_shift($view->args);
+    $script .= "\nwindow._ti['internalSearch'] = '$search_key'";
+    $script .= "\nwindow._ti['numberSearchResults'] = '$count';";
+
+    if (!empty($view->human_name)) {
+      $script .= "\nwindow._ti['pageTitle'] = '$view->human_name';";
+    }
```

Comment:
`ding_webtrekk_page_alter`, `ding_webtrekk_panels_pane_content_alter` and `ding_webtrekk_views_post_execute` all share a similar pattern á la 

```js
window._ti = window._ti || {};
window._ti['key 1'] = 'value A';
window._ti['key 2'] = 'value B';
// etc.
```

This sort of strings-as-code is error prone - missing characters, strings in strings etc. It is also duplicated. The intention is to expand the use of webtrekk in the system so this would spread throughout more hooks etc. That would only make the situation worse.

Please consider whether it could be wrapped in a function á la:

```php
function _ding_webtrekk_set_datalayer_values(array $value_map) {
  $script_lines = array_map(function($key, $value) {
    return "window._ti['$key'] = '$value'"; 
  }, array_keys($value_map), $value_map);
  $script_lines = array_merge (
    [ 'window._ti = window._ti || {}' ],
    $script_lines
  );

  $script = implode(";\n", $script_lines);
  drupal_add_js($script, array(
   'type' => 'inline',
  ));
}
```

NB: Code above is only a proff of concept - it is totally untested.